File: php/StorableObject.php
<?php

/**
 * A base class for storable objects
 *
 * @author paul
 */
abstract class StorableObject {
  private $_additional_vars = array();

  abstract function get_id();

  abstract function set_id($id);

  abstract function get_name();

  abstract function to_json();

  abstract function from_json($json);

  public function __get($name) {
    return $this->_additional_vars[$name];
  }

  public function __set($name, $value) {
    $this->_additional_vars[$name] = $value;
  }

  public function to_array() {
    $arr = get_object_vars($this);
    foreach ($this->_additional_vars as $name => $value) {
      $arr[$name] = $value;
    }
    unset($arr['_additional_vars']);
    return $arr;
  }
}

?>

File: java/lib/src/main/java/com/caffeinatedbliss/kayveep/StorableObject.java
package com.caffeinatedbliss.kayveep;

/**
 * @author Paul S. Hawke (paul.hawke@gmail.com)
 *         On: 2/22/12 at 11:57 PM
 */
public interface StorableObject {
	String getId();

	void setId(String id);

	String getName();

	void setName(String name);

	String toJson();

	Object get(String name);

	void set(String name, Object value);
}
